CMakeLists.txt

```diff
@@ -323,6 +319,48 @@ if(IREE_ERROR_ON_MISSING_SUBMODULES AND Python3_FOUND AND Git_FOUND)
   endif()
 endif()
 
+#-------------------------------------------------------------------------------
+# CUDA configuration for both the compiler and runtime.
+# We do this at the top level so that we can fail fast and make global
+# decisions that effect both compiler and runtime. It also helps with error
+# messaging to do this all in one place, since we can provide very targeted
+# advice.
+#-------------------------------------------------------------------------------
+
+set(IREE_CUDA_LIBDEVICE_PATH "" CACHE STRING "Absolute path to an appropriate libdevice.*.bc (needed to build the IREE cuda compiler target)")
+
+# If any CUDA features are being built, try to locate a CUDA SDK. We will fall
+# back to this as needed for specific features.
+if(IREE_TARGET_BACKEND_CUDA OR IREE_HAL_DRIVER_CUDA)
+  find_package(CUDAToolkit QUIET)
```

Comment:
note that we could put a version in here: https://cmake.org/cmake/help/latest/module/FindCUDAToolkit.html#arguments

(if we know that we need at least a specific version, or one exact version)

Reply:
Yes, I don't actually know such things so will leave it to the more informed to make a decision.